Providers/Profile.swift

```diff
@@ -503,44 +498,40 @@ public class BrowserProfile: Profile {
         self.remoteClientsAndTabs.insertCommands(commands, forClients: clients) >>> { self.syncManager.syncClients() }
     }
 
-    lazy var logins: protocol<BrowserLogins, SyncableLogins, ResettableSyncStorage> = {
+    lazy var logins: BrowserLogins & SyncableLogins & ResettableSyncStorage = {
         return SQLiteLogins(db: self.loginsDB)
     }()
 
     // This is currently only used within the dispatch_once block in loginsDB, so we don't
     // have to worry about races giving us two keys. But if this were ever to be used
     // elsewhere, it'd be unsafe, so we wrap this in a dispatch_once, too.
-    private var loginsKey: String? {
+    lazy private var loginsKey: String? = {
```

Comment:
We've talked about this before, no? The context was @thebnich's work (IIRC) around shutdown crashes and reopening DBs. My conclusion was that `Profile` was a singleton in disguise: it's not safe to create two instances in the same process, because they use `dispatch_once` (and some statics) to manage the lifecycle of their members.

We didn't use `lazy` in the Swift 1 days because it wasn't thread-safe. Perhaps that's changed?

The correct course of action would seem to be to de-singleton `Profile`: turn these into (thread-safe) per-instance accessors. If that can't be done for some reason, or introduces too much risk, then make these a correct singleton and leave a comment.